### ISSUE TITLE
Beta Test Feedback Day 1

### ISSUE THREAD

**Issue by pulkitsinghal** [ShoppinPal/warehouse#158]
After each task gets logged and split into individual issues with each one in some sort of priority order relative to the other ... feel free to mark this issue as done. Its only goal in life is to help other issues be created and prioritized.

**Comment by pulkitsinghal** [comment]
* Blockers for warehouse beta testing:
  1. bug - order qty and fulfilled qty are 1 and 1 but show up as 0 and 1
  2. filter out anything with ordered quantity of zero
  3. need to know what's in the box for the warehouse folks

**Comment by pulkitsinghal** [comment]
* likely to cause a lot of duplicates without improved UX/UI:
 1. viewable comments should not be green on green, please change the color of the comment text ... and also increase the font size so that its readable
 1. for generated rows that are under progress say: "Generating ... you may tap to check status every few minutes"
 1. done button inside the edit row to put it away without the need to dismiss and then swipe it to completed ... should be on the right
 1. delete button inside the edit row for rows we just don't want to deal with ... should be on the left ... otherwise order quantities need to be explicitly zero-ed out (lots of extra taps) and afterwards, dummy rows with no purpose travel through the entire workflow
* manager's feedback
 1. how do we know how much work is left for the manager to do ... some kind of progress bar or 10/100 or 90 left or something
* future facing
  1. file issues describing the process of handling negative quantities by adding buttons for `shoplifting` and `damaged goods` etc. to the workflow
* confusing
 1. take out the last field from the generate order screen
* sorting related
  1. No headers for the warehouse report
  1. Add sorting for warehouse:
    1. once the values are tracked as tags, they will either be numbered from 1-1000
    1. or, be numbered as 001002003 based on aisle(001)-bin(002)-box(003) which is the warehouse admin's responsibility to get right ... but no more additional code for us when sorting ... i hope?
  1. vibrating and nonVibrating departments should show up on top for businesses like Patricias as its their bread & butter ... give some thought to more types of sorting structures to help make this happen ... alphabetical is too simple
* very minor
  1. the waiting spinner should be fixed so that you can't scroll away and forget why your screen is locked
* data integrity
 1. cost should be baked into the rows when the report is generated

**Comment by elbrando** [comment]
Of the items unchecked, the order should be:
1a) give +/- signs for the DSL field (all numerical fields, not even worrying about look) too
1b) done button inside the edit row to put it away without the need to dismiss and then swipe it to completed ... should be on the right
1c) delete button inside the edit row for rows we just don't want to deal with ... should be on the left ... otherwise order quantities need to be explicitly zero-ed out (lots of extra taps) and afterwards, dummy rows with no purpose travel through the entire workflow
2) Managers comments are difficult on Warehouse side. When swiped left to edit, they disappear from view. The whole "comments" thing can be revisited at a lower priority. I've got some ideas.
3) Add sorting for warehouse:  be numbered as 001002003 based on aisle(001)-bin(002)-box(003) which is the warehouse admin's responsibility to get right ... but no more additional code for us when sorting ... i hope? <-- yes, your hope should be confirmed.
4) Department sorting thing. I'm open to solutions. Tags maybe?
5) "Addons" work around seems problematic to me. I'd like to see it bumped up. Especially because we will need this ability in order to do in-store manual orders.
6) "No headers for the warehouse report" -- Leave headers in. They actually helped. May need to revisit this or at least figure out how it would work with BinLoc.

Everything above needs to be done before I can launch it across the organization. Everything else can be done as needed (I think).

Things I don't understand for various reasons:
1) if the user's role is manager then do not popup the virtual keyboard on the iPad
2) take out the last field from the generate order screen

The fixes were terrific. Especially the "Generating..." part. The boxes thing worked out well enough. We can "fix" it later.

**Comment by pulkitsinghal** [comment]
@srisub - please see if you can get the clarion team to do the UI/layout work for `1a,b,c`, I can take care of wiring it to the backend and adding any logical stuff to the UI.

@elbrando - as discussed, I'll try to wrap up (1) and (2) from the previous comment by EOD tomorrow.

**Comment by pulkitsinghal** [comment]
@srisub - due to some reasons that change the scenario to be one where managers and warehouse folks will *NOT* have IR scanners hooked to their iPad for several months to come ... @elbrando agreed to let us remove virtual keyboard across the board from warehouse and give everyone the benefit of a simpler experience and better speed with the native iOS keyboard.

I'm not executing on this right away but just leaving an informative comment about it here for now.

**Comment by pulkitsinghal** [comment]
The folks in the warehouse should now be able to see the manager's comments and will not be prompted to enter their own in `edit` mode for a row ... but @srisub the styling, layout , font size and font color still suck supremely ... so can you get someone to tweak that from your team.

**Comment by pulkitsinghal** [comment]
@elbrando - we should revisit this and look over the remaining items and move them into a separate bucket and close this one out?
cc @srisub
